sdk/core/Azure.Core.TestFramework/src/Azure.Core.TestFramework.csproj

```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Azure.Core" />
-    <PackageReference Include="Azure.Identity" />
+    <PackageReference Include="Azure.Identity" Version="1.10.3" />
```

Comment:
We should update the central package.data.props file instead.

Reply:
![image](https://github.com/Azure/azure-sdk-for-net/assets/67986960/ad387e04-bc23-4d22-8a78-d55931182e45)
I think Azure.Identity is 1.10.3 already in Packages.Data.props but for some reason this version is not being used in some SDKs

Reply:
The projects you are update here are test projects. There is a separate dependency for those that looks like it was already updated https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L223. So it looks like the issue this PR is attempting to fix is already resolved.

Reply:
oh that's interesting - maybe we should close the issue then? @pallavit  could you please take a look?

Reply:
Hey - I think the component governance issue is complaining about other dependencies that we have which are not necessarily using pacakge.data.props - https://grep.app/search?q=Azure.Identity&filter[repo][0]=Azure/azure-sdk-for-net&filter[lang][0]=XML

Note some of these are in samples - and are not user scenarios but component governance flags on source browse so they do get flagged.

@nisha-bhatia please take a look at the projects which are being asked to be upgraded in the CG alert and check - I believe a lot of those are not dependent on package.data.props.

Reply:
For some cross-package, complete Key Vault samples I have to reference package versions conditionally because they could be built outside our repo. It's possible that is the source of some of these.

Reply:
The governance alert last run in October, which explains why it hasn't been resolved - https://dev.azure.com/azure-sdk/internal/_componentGovernance/98522/alert/9831951?typeId=7614213&pipelinesTrackingFilter=1. The packages it references have already been upgraded. It can probably be dismissed.

Reply:
> That said, we should still update any other instances to use a non-deprecated version.

Yeah CG alert is stale and can be dismissed but we did have references to older packages from my basic grep so I wanted to make sure we update those before we dismiss the CG alert for completeness.